core/templates/dev/head/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.directive.ts

```diff
@@ -66,57 +67,68 @@ oppia.directive('adminDevModeActivitiesTab', [
+              return;
+            }
+            if (
+              !$window.confirm('This action is irreversible. Are you sure?')) {
+              return;
+            }
 
-          var numSucceeded = 0;
-          var numFailed = 0;
-          var numTried = 0;
-          var printResult = function() {
-            if (numTried < GLOBALS.DEMO_EXPLORATION_IDS.length) {
+            ctrl.setStatusMessage('Processing...');
+            AdminTaskManagerService.startTask();
+
+            var numSucceeded = 0;
+            var numFailed = 0;
+            var numTried = 0;
+            var printResult = function() {
+              if (numTried < demoExplorationIds.length) {
+                ctrl.setStatusMessage(
+                  'Processing...' + numTried + '/' +
+                  demoExplorationIds.length);
+                return;
+              }
               ctrl.setStatusMessage(
-                'Processing...' + numTried + '/' +
-                GLOBALS.DEMO_EXPLORATION_IDS.length);
-              return;
+                'Reloaded ' + demoExplorationIds.length +
+                ' explorations: ' + numSucceeded + ' succeeded, ' + numFailed +
+                ' failed.');
+              AdminTaskManagerService.finishTask();
+            };
+
+            for (var i = 0; i < demoExplorationIds.length; ++i) {
+              var explorationId = demoExplorationIds[i];
+
+              $http.post(ADMIN_HANDLER_URL, {
+                action: 'reload_exploration',
+                exploration_id: explorationId
+              }).then(function() {
+                ++numSucceeded;
+                ++numTried;
+                printResult();
+              }, function() {
+                ++numFailed;
+                ++numTried;
+                printResult();
+              });
             }
-            ctrl.setStatusMessage(
-              'Reloaded ' + GLOBALS.DEMO_EXPLORATION_IDS.length +
-              ' explorations: ' + numSucceeded + ' succeeded, ' + numFailed +
-              ' failed.');
-            AdminTaskManagerService.finishTask();
-          };
-
-          for (var i = 0; i < GLOBALS.DEMO_EXPLORATION_IDS.length; ++i) {
-            var explorationId = GLOBALS.DEMO_EXPLORATION_IDS[i];
-
-            $http.post(ADMIN_HANDLER_URL, {
-              action: 'reload_exploration',
-              exploration_id: explorationId
-            }).then(function() {
-              ++numSucceeded;
-              ++numTried;
-              printResult();
-            }, function() {
-              ++numFailed;
-              ++numTried;
-              printResult();
-            });
           }
         };
 
+        AdminDataService.getDataAsync().then(function(response) {
+          ctrl.DEMO_EXPLORATIONS = response.demo_explorations;
+          ctrl.DEMO_COLLECTIONS = response.demo_collections;
+          demoExplorationIds = response.demo_exploration_ids;
+          canReloadAllExplorations = true;
```

Comment:
Do we have to do such changes in the frontend to disable the button until this value is true?

Reply:
I don't think this is necessary since the check is already done in the function.

Reply:
On second thoughts, I think this is okay.
Done. Thanks